src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -246,7 +246,7 @@ trait TypeOps { this: Context => // TODO: Make standalone object.
     case AndType(l, r) =>
       simplify(l, theMap) & simplify(r, theMap)
     case OrType(l, r) =>
```

Comment:
I think this is OK, because if we call `simplified` on a type not written by a user which contains an `OrType`, then this `OrType` must have been constructed in the code by calling `lub`, so this `OrType` cannot have a singleton in it. And if we do not keep singletons in `simplified`, then we will always lose them because we call `simplified` in `Typer#adapt`.